clients/geth/specular/rollup/derivation/batch_builder.go

```diff
@@ -20,19 +20,18 @@ type batchBuilder struct {
 	lastAppended types.BlockID
 }
 
+type BlockAttributes struct {
+	Timestamp *big.Int
+	Txs       [][]byte // encoded batch of transactions.
+}
+
 type BatchAttributes struct {
-	contexts           []*big.Int
-	txLengths          []*big.Int
-	firstL2BlockNumber *big.Int
-	txBatchVersion     *big.Int
-	txBatch            []byte // encoded batch of transactions.
+	FirstL2BlockNumber *big.Int
+	Blocks             []BlockAttributes
```

Comment:
The RLP library uses reflection for encoding so I had to change these fields to public.